daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -1840,6 +1840,14 @@ func (devices *DeviceSet) initDevmapper(doInit bool) error {
 		if err := devicemapper.CreatePool(devices.getPoolName(), dataFile, metadataFile, devices.thinpBlockSize); err != nil {
 			return err
 		}
+		defer func() {
+			if retErr != nil {
```

Comment:
Does this code really work? I see that retErr is named return variable but nobody is operating on that variable. So it should always be 0 and defer function will not do anything?

Or I am understanding it wrong.